zenoh/src/value.rs

```diff
@@ -36,23 +36,36 @@ impl Value {
             encoding: Encoding::default(),
         }
     }
-
     /// Creates an empty [`Value`].
     pub const fn empty() -> Self {
         Value {
             payload: Payload::empty(),
             encoding: Encoding::default(),
         }
     }
+    /// Checks if the [`Value`] is empty.
+    /// Value is considered empty if its payload is empty and encoding is default.
+    pub fn is_empty(&self) -> bool {
+        self.payload.is_empty() && self.encoding == Encoding::default()
+    }
+}
 
-    /// Sets the encoding of this [`Value`]`.
-    #[inline(always)]
-    pub fn with_encoding<IntoEncoding>(mut self, encoding: IntoEncoding) -> Self
-    where
-        IntoEncoding: Into<Encoding>,
-    {
-        self.encoding = encoding.into();
-        self
+impl ValueBuilderTrait for Value {
```

Comment:
`Value` has not been ported yet to accessor pattern.
By implementing `ValueBuilderTrait` directly on `Value` we will have a conflict in terms of functions.

With accessor pattern:
```
let payload: &Payload = value.payload();
```

With builder pattern:
```
value = value.payload("foo");
```

Do we really need to implement `ValueBuilderTrait` on `Value` itself or could we have a `ValueBuilder` as we do with a `SampleBuilder`?